tools/make/lint.mk

```diff
@@ -4,27 +4,56 @@
 
 ##@ Lint
 
+GITHUB_ACTION ?=
+
 .PHONY: lint
 lint: ## Run all linter of code sources, including golint, yamllint, whitenoise lint and codespell.
-	@$(MAKE) lint.golint lint.yamllint lint.codespell lint.whitenoise
 
+PHONY: lint-deps
+lint-deps: ## Everything necessary to lint (useful to separate out in the logs)
+
+GOLANGCI_LINT_FLAGS ?= $(if $(GITHUB_ACTION),--out-format=github-actions)
 .PHONY: lint.golint
-lint.golint:
+lint: lint.golint
+lint-deps: $(tools/golangci-lint)
+lint.golint: $(tools/golangci-lint)
 	@echo Running Go linter ...
-	@golangci-lint run --build-tags=e2e --config=tools/linter/golangci-lint/.golangci.yml
+	$(tools/golangci-lint) run $(GOLANGCI_LINT_FLAGS) --build-tags=e2e --config=tools/linter/golangci-lint/.golangci.yml
 
 .PHONY: lint.yamllint
-lint.yamllint:
+lint: lint.yamllint
+lint-deps: $(tools/yamllint)
+lint.yamllint: $(tools/yamllint)
 	@echo Running YAML linter ...
-	@yamllint --config-file=tools/linter/yamllint/.yamllint changelogs/
+	$(tools/yamllint) --config-file=tools/linter/yamllint/.yamllint changelogs/
 
+CODESPELL_FLAGS ?= $(if $(GITHUB_ACTION),--disable-colors)
 .PHONY: lint.codespell
+lint: lint.codespell
+lint-deps: $(tools/codespell)
 lint.codespell: CODESPELL_SKIP := $(shell cat tools/linter/codespell/.codespell.skip | tr \\n ',')
-lint.codespell:
+lint.codespell: $(tools/codespell)
 	@echo Running Codespell linter ...
-	@codespell --skip $(CODESPELL_SKIP) --ignore-words tools/linter/codespell/.codespell.ignorewords --check-filenames --check-hidden -q2
+# This ::add-matcher/::remove-matcher business is based on
+# https://github.com/codespell-project/actions-codespell/blob/2292753ad350451611cafcbabc3abe387491339a/entrypoint.sh
+# We do this here instead of just using
+# codespell-project/codespell-problem-matcher@v1 so that the matcher
+# doesn't apply to the other linters that `make lint` also runs.
```

Comment:
As mentioned in the commit message (ec2c0181debc270e51139f5b64b6d23c72c8960a), before #113 it wasn't necessary because the linter was being run by the `codespell-project/actions-codespell` action rather than by the Makefile; when #113 moved it to be done by the Makefile, it dropped this piece of functionality.  ec2c01 is simply updating the Makefile to do what the `codespell-project/actions-codespell`  action was doing.

Reply:
thanks for the bg, this looks like something that will be hard to maintain in the future, but it is simply readding a  check that was removed so resolving this